src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+  const copyState = { ...state };
+
+  for (const middleActions of actions) {
+    switch (middleActions.type) {
+      case 'addProperties':
+        Object.assign(copyState, middleActions.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const elem of middleActions.keysToRemove) {
+          delete copyState[elem];
+        }
+        break;
+
+      case 'clear':
+        for (const key in copyState) {
+          delete copyState[key];
+        }
```

Comment:
```suggestion
     copyState = {}
```

Reply:
why didn't you fix this?
it will use less resources in this case